### PR TITLE
Ensure all field types of SKYRecord conform NSCopying and NSCoding

### PR DESCRIPTION
**Pull request by ben181231**
connect #144

### CODE REVIEW

File: Example/Tests/SKYReferenceTests.m [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+#import <Foundation/Foundation.h>
+#import <OHHTTPStubs/OHHTTPStubs.h>
```
Comment (cheungpat) [inline Example/Tests/SKYReferenceTests.m]: no need for this header for this file

File: Example/Tests/SKYReferenceTests.m [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+        it(@"can be encoded and decoded", ^{
+            SKYReference *ref = [[SKYReference alloc]
+                                 initWithRecordID:[SKYRecordID recordIDWithRecordType:@"book"]];
+
+            NSData *data = [NSKeyedArchiver archivedDataWithRootObject:ref];
+            SKYReference *refClone = [NSKeyedUnarchiver unarchiveObjectWithData:data];
+
+            //// Don't know why this cannot pass
+            // expect(refClone.recordID).to.equal(ref.recordID);
```
Comment (cheungpat) [inline Example/Tests/SKYReferenceTests.m]: I don’t know too, if I am not mistaken, the runtime consider the two objects have different classes, causing the the equality comparison to fail. This happens after archiving.

File: Pod/Classes/SKYReference.m [hunk 3]
```diff
@@ -111,7 +111,8 @@ - (NSUInteger)hash
 
 - (instancetype)initWithCoder:(NSCoder *)decoder
 {
-    SKYRecordID *recordID = [decoder decodeObjectForKey:@"recordID"];
+
```
Comment (cheungpat) [inline Pod/Classes/SKYReference.m]: this extra blank line is not needed

File: Example/Tests/SKYReferenceTests.m (hunk 2 above)
Comment (ben181231) [reply Example/Tests/SKYReferenceTests.m]: > the runtime consider the two objects have different classes
That is also what I get during my debugging. However, both `[refClone.recordID class]` and `[ref.recordID class]` return `SKYRecordID` when I print them in lldb console.

Comment (cheungpat) [reply Example/Tests/SKYReferenceTests.m]: the object’s `isa` field is different, god knows why

**Comment by ben181231** [comment]
@cheungpat the PR updated

**Pull request merged**